src/coreclr/jit/block.h

```diff
@@ -882,9 +882,14 @@ struct BasicBlock : private LIR::Range
     void ensurePredListOrder(Compiler* compiler);
     void reorderPredList(Compiler* compiler);
 
-    BlockSet    bbReach; // Set of all blocks that can reach this one
-    BasicBlock* bbIDom;  // Represent the closest dominator to this block (called the Immediate
-                         // Dominator) used to compute the dominance tree.
+    BlockSet bbReach; // Set of all blocks that can reach this one
+
+    union {
+        BasicBlock* bbIDom; // Represent the closest dominator to this block (called the Immediate
+                            // Dominator) used to compute the dominance tree.
+        void* bbProbeList;  // Used early on by fgInstrument
```

Comment:
Do you have something that nulls these out before the phases that uses them? Or initializes them before they are read? I see somewhere that you assert bbInfo is != nullptr, for instance.

Reply:
For instrumentation this happens in the `VisitBlock` method during the tree walk. For reconstruction the value is set during `Prepare` before it is read.